### PR TITLE
Add currency check to acceptance test

### PR DESCRIPTION
**Pull request by mario-galic**
# Why are you doing this?
In order to avoid the same type of bug in the future (https://github.com/guardian/membership-frontend/pull/1434), we added an extra check in the membership frontend's acceptance test.

# Changes
* Check currency can change to non-UK currency
* Switch to Supporter flow from Partner flow

### CODE REVIEW

File: frontend/test/acceptance/JoinPartnerSpec.scala [hunk 1]
```diff
@@ -72,7 +78,7 @@ class JoinPartnerSpec extends FeatureSpec with Browser
       assert(ThankYou.pageHasLoaded)
 
       And("should be signed in as Partner.")
```
Comment (rtyley) [inline frontend/test/acceptance/JoinPartnerSpec.scala]: 'Partner' should be 'Supporter' now!

File: frontend/test/acceptance/JoinPartnerSpec.scala [hunk 2]
```diff
@@ -59,6 +59,12 @@ class JoinPartnerSpec extends FeatureSpec with Browser
       And("should be logged in with their Identity account.")
       assert(enterDetails.userIsSignedIn)
 
+      When("users select non-UK country,")
+      enterDetails.changeCountry()
+
+      Then("the currency should change.")
```
Comment (rtyley) [inline frontend/test/acceptance/JoinPartnerSpec.scala]: I guess the default currency is GBP, so the currency _will_ have changed? We don't seem to be specifying a `countryGroup` when we go to `/join/supporter/enter-details` - so I'm not sure if there's any chance the currency might already be USD, before the country is set to US.

File: frontend/test/acceptance/pages/EnterDetails.scala [hunk 3]
```diff
@@ -25,6 +25,10 @@ case class EnterDetails(val testUser: TestUser) extends Page with Browser {
 
   def fillInCardDeclinedProcessError() { CreditCard.fillInCardDeclinedProcessError() }
 
+  def changeCountry() { DeliveryAddress.selectUSA() }
```
Comment (rtyley) [inline frontend/test/acceptance/pages/EnterDetails.scala]: `changeCountry()` is a bit vague here, I think you're better off making this `changeCountryTo("US")` - and then you could change country a couple of times, to checking currency each time.

File: frontend/test/acceptance/JoinPartnerSpec.scala (hunk 1 above)
Comment (svillafe) [inline frontend/test/acceptance/JoinPartnerSpec.scala]: Also the file should be called JoinSupporterSpec.scala

**Comment by rtyley** [comment]
:+1:

**Comment by svillafe** [comment]
It is working

**Comment by svillafe** [comment]
It is working on PROD

**Pull request merged**